Automattic/newspack-ads
# chore: create suppression class and functions.php file

This PR is a part of #333, creating a namespaced `Suppression` class in accordance with the plugin's overall structure and moves the `newspack_ads_should_show_ads()` public function to an unnamespaced file dedicated for such functions.

Next steps for future PR(s) should be:
 - Move `get_suppression_config()` and global suppression config away from `Providers\GAM_Model`
 - Create API for modifying global suppression config
 - Modify `newspack-plugin` to use this API

### How to test

No change should take place:

 - Change the global suppression config in the Newspack Ads Suppression wizard and confirm they are applied
 - Edit a post, toggle the "Don't show ads on this post or page" and confirm it is applied
